### ISSUE TITLE
New Model: optimized and ultimed model variants of parakeet v3

### ISSUE THREAD

**Issue by thiswillbeyourgithub** [Kieirra/murmure#448]
### Murmure version

Any

### Operating system

Linux

### Display server (Linux only)

None

### Feature impacted

Transcription

### Bug description

Hi,

Finally pushed my models. I made a french medical dataset I called UltiMed-v1, finetuned parakeet on it, learned many tweaks along the way to create a brand new optimized onnx model to sunset the smoothquant version.

Here are links to the entire thing:

The dataset UltiMed-ASR-FR-V1
https://huggingface.co/datasets/Olicorne/UltiMed-ASR-FR-v1

The UltiMed finetuned:
https://huggingface.co/Olicorne/parakeet-tdt-0.6b-v3-UltiMed-onnx

The optimized model (better than smoothquant):
https://huggingface.co/Olicorne/parakeet-tdt-0.6b-v3-optimized-onnx

The scripts I used to make the dataset:
https://github.com/thiswillbeyourgithub/UltiMed-ASR-FR-v1-scripts

The training scripts:
https://github.com/thiswillbeyourgithub/UltiMed-ASR-FR-v1-NeMo_training_scripts

The voxtral instance I used to generate the audio:
https://github.com/thiswillbeyourgithub/UltiMed-ASR-FR-v1-Voxtral

So my questions:
1. would you mind giving the optimized onnx a try? There are all better than smoothquant and can be much more compact and/or require less ram for the model and its activations. AFAIK they are pareto improvements.
2. would you mind giving a try to the UltiMed finetuned variant on murmure please? On my benchmarks I'm getting 7-8x less errors on medical stuff, but haven't been able to test in real usage. The loss on west european language should be minimal.
3. could you document how to swap onnx models on murmure so I could ask a few murmure users I know how to swap it? Or, if possible, add an advanced setting to swap model? It would require making sure murmure can work with all variants I'm supplying.

Note: some of the benchmarks are still being recomputed, just in case. I'll update the model card.

Very interested in all feedbacks! This is only the V1 variant.

### Steps to reproduce

.

### Debug logs

```shell

```

**Comment by Kieirra** [comment]
Nice, thanks a lot.

I'll test it and let you know. Since I'll be on vacation in a few days without a computer, it may take me a little while to get back to you. I’m not sure I’ll be able to test it beforehand.

**Comment by Kieirra** [comment]
OK so I tested the optimized model against the previous lite model (in production).

On my dataset both models return the same WER and CER. I also used it for a few days of daily work and could not really tell them apart.

The difference is RAM. Same release build, measured on the running process after a dictation: 
- about 1.0 to 1.4 go resident with lite model, 
- about 1.4 to 1.7 go with the optimized model, and it stays there. 

So on my side, no visible benefit yet and a heavier model.

Regarding UltiMed, I can build a Murmure version with it, but not being in the medical field, I can't judge if it's really better. The simplest way is to send me a few real Murmure recordings (with "Keep audio recordings" enabled) on medical sentence to see if there's a noticeable difference.

<img width="884" height="389" alt="Image" src="https://github.com/user-attachments/assets/678056ec-0ade-4bdc-a466-de155a9bc51b" />

**Comment by thiswillbeyourgithub** [comment]
Thanks for giving it a try.

Can you tell me which quant of the ultimed finetuned you used? And do all variants work out of the box on murmure or only int8?

**Comment by Kieirra** [comment]
I used the optimized model, int8 variant: https://huggingface.co/Olicorne/parakeet-tdt-0.6b-v3-optimized-onnx

Only int8 works out of the box. Murmure runs on CPU only, so that is the only variant it loads.

For UltiMed, like I said, I can't really test it myself. I don't have the medical background to dictate realistic sentences and judge if the output is right. Since it is a model for a specific use, the feedback has to come from medical users but I can test it on real sentences recorded with Murmure and compare it with the vanilla model to see if it is better or not.

**Comment by thiswillbeyourgithub** [comment]
Thanks. Claude somehow found a way to frankenstein the int8-lite and the new int8 to optimize disk and RAM. I just pushed it. Can you give it a try?

**Comment by thiswillbeyourgithub** [comment]
Also, claude had a suggestion for murmure dev. I asked it to condense it to explore the idea yourself with your LLM as I'm not familiar with your codebase : 


> One-liner for the murmure dev

> In src-tauri/src/engine/engine.rs we use ort 2.0.0-rc.10: add a RunOptions with config entry "memory.enable_memory_arena_shrinkage" = "cpu:0" and call run_with_options for the ENCODER session only (never the per-token  decoder, which runs thousands of times per utterance), then benchmark load  RSS, post-run RSS and wall-clock per run over a 30s and a 10min  transcription in three configs (no change / arena shrinkage on encoder /  CPU arena disabled entirely at session build) and report the table.

> Worth telling them the target: on my probe the arena never returns ~590 MB of high-water mark (1806 MB steady with the arena on, 1214 MB with it off, same model, same input).

**Comment by Kieirra** [comment]
> Thanks. Claude somehow found a way to frankenstein the int8-lite and the new int8 to optimize disk and RAM. I just pushed it. Can you give it a try?

The new int8 is better than the previous one, around 1.5 GB in RAM and 200 MB smaller on disk. In RAM it is still about 100 MB above the lite we ship today, and RAM is our constraint, not disk. WER and CER are identical..

> Also, claude had a suggestion for murmure dev. I asked it to condense it to explore the idea yourself with your LLM as I'm not familiar with your codebase :

Arena is disabled by default in the ort crate I use, not sure why he's giving you this tip.

**Comment by thiswillbeyourgithub** [comment]
Would you mind trying 1 or 2 audios of length 0-10s, then 10-20, 20-30, 30-40, 40-50? I have a hunch one of the models uses less ram on short audios and the other one on long audios. And I'm not sure of the cutoff but I suspect around 30s.

**Comment by Kieirra** [comment]
I redid everything with a fresh process per audio because I think I mixed up a bit my last test on the new-int8. 

| Audio length | RAM lite | RAM new int8 | 
|---|---|---|
| 0-10 s | 1526 MB | 1289 MB | 
| 10-20 s | 1523 MB | 1265 MB | 
| 20-30 s | 1524 MB | 1298 MB |
| 30-40 s | 1542 MB | 1301 MB |
| 40-50 s | 1519 MB | 1323 MB |

Quality on my full corpus, 31 sentences and 1231 words, same binary:

| | WER | WER with dictionary | CER | CER with dictionary |
|---|---|---|---|---|
| lite | 4.39 % | 4.22 % | 2.29 % | 2.39 % |
| new int8 | 4.55 % | 4.71 % | 2.53 % | 2.63 %

So it seems the new int8 is lighter in RAM and not that bad in WER/CER.

**Comment by thiswillbeyourgithub** [comment]
And has no time limits either, and probably is lesd french biaised.
